sql-plugin/src/main/scala/org/apache/spark/sql/execution/joins/rapids/CpuSortMergeJoinFallback.scala

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.joins.rapids
+
+import com.nvidia.spark.rapids.{ColumnarToRowIterator, GpuOutOfCoreSortIterator, RowToColumnarIterator}
+import com.nvidia.spark.rapids.RapidsBuffer.SpillCallback
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.{Ascending, Attribute, Expression, JoinedRow, Predicate, Projection, RowOrdering, SortOrder, UnsafeProjection, UnsafeRow}
+import org.apache.spark.sql.catalyst.plans.{InnerLike, JoinType}
+import org.apache.spark.sql.execution.{ExternalAppendOnlyUnsafeRowArray, RowIterator}
+import org.apache.spark.sql.execution.joins.SortMergeJoinScanner
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+class CpuSortMergeJoinFallback(
+    val joinType: JoinType,
+    val left: Iterator[ColumnarBatch],
+    val leftSchema: Seq[Attribute],
+    val unboundLeftGpuKeys: Seq[Expression],
+    val unboundLeftCpuKeys: Seq[Expression],
+    val right: Iterator[ColumnarBatch],
+    val rightSchema: Seq[Attribute],
+    val unboundRightGpuKeys: Seq[Expression],
+    val unboundRightCpuKeys: Seq[Expression],
+    val cpuCondition: Option[Expression],
+    val outputSchema: Seq[Attribute],
+    private val targetSize: Long,
+    private val spillThreshold: Int,
+    private val inMemoryThreshold: Int,
+    private val spillCallback: SpillCallback,
+    private val cleanupResources: () => Unit) extends Iterator[ColumnarBatch] {
```

Comment:
In practice `cleanupResources` seems to be a no-op always?  Maybe I misread something.